src/components/component/secrets/secret-list-item-title-azure-key-vault-item.tsx

```diff
@@ -12,6 +12,27 @@ import { RequestState } from '../../../state/state-utils/request-states';
 
 import '../style.css';
 
+function getSecretInReplicaWithUniqueBatchNames(secretInReplicaList) {
+  let batchMap = new Map<string, Map<string, boolean>>(); //batchName:version:boolean
+  // this filtering is to avoid showing all job pods for the same batch with the same secret version
+  return secretInReplicaList.filter((secretInReplica) => {
+    if (
+      secretInReplica.batchName === null ||
+      secretInReplica.batchName.length === 0
+    )
+      return true;
+    if (!batchMap.has(secretInReplica.batchName))
+      batchMap.set(secretInReplica.batchName, new Map<string, boolean>());
+    if (!batchMap.get(secretInReplica.batchName).has(secretInReplica.version)) {
+      batchMap
+        .get(secretInReplica.batchName)
+        .set(secretInReplica.version, true); // first version for this batch
+      return true;
+    }
+    return false;
+  });
+}
```

Comment:
function could be simplified using `reduce()`
```suggestion
function filterSecrets(
  secrets: Array<Readonly<AzureKeyVaultSecretStatusModel>>
): Array<AzureKeyVaultSecretStatusModel> {
  // avoid showing duplicate secrets for job pods with same batchName and version
  return secrets?.reduce<Array<AzureKeyVaultSecretStatusModel>>((obj, x) => {
    if (
      !(x.batchName?.length > 0) ||
      !obj.find((y) => y.batchName === x.batchName && y.version === x.version)
    ) {
      obj.push(x);
    }
    return obj;
  }, []);
}
```

Reply:
Applied